pyccel/codegen/printing/ccode.py

```diff
@@ -1632,10 +1636,22 @@ def _print_PyccelUnarySub(self, expr):
         return '-{}'.format(self._print(expr.args[0]))
 
     def _print_AugAssign(self, expr):
-        lhs_code = self._print(expr.lhs)
         op = expr.op
-        rhs_code = self._print(expr.rhs)
-        return "{0} {1}= {2};\n".format(lhs_code, op, rhs_code)
+        lhs = expr.lhs
+        rhs = expr.rhs
+        lhs_code = self._print(lhs)
+        rhs_code = self._print(rhs)
+        if op == '%':
+            # Two cases where this works: (NativeFloat and NativeFloat), (NativeFloat and NativeInt)
+            # The case (NativeInt and NativeFloat) would cause a type change error.
+            if isinstance(lhs.dtype, NativeFloat) or isinstance(rhs.dtype, NativeFloat):
+                self.add_import(c_imports['math'])
+                # Get (dtype, precision) pair for dtype_registry
+                key  = (str(lhs.dtype), get_final_precision(lhs))
+                # Get the proper modulo function (fmod for doubles, fmodf for floats)
+                func = modulo_fmod[dtype_registry[key]]
+                return "{0} = {1}({0}, {2});\n".format(lhs_code, func, rhs_code)
```

Comment:
Does this code duplicate what is in `_print_PyccelMod`? If it doesn't what is the difference and why is there is a difference?
If it does duplicate then this is dangerous practice (see [DRY principle](https://thevaluable.dev/dry-principle-cost-benefit-example/)). If anyone changes `_print_PyccelMod` in the future then they are likely to forget to also change this bit of code. Why not use something similar to Fortran's [`_print_AugAssign`](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L2066-L2068) to handle the modulo case?

Reply:
Ok, I didn't know `pyc_modulo` and `pyc_fmodulo` exist already. That renders all changes I did useless 😄.

Reply:
:laughing: that sounds like a good argument for mimicking Fortran's implemenation of `_print_AugAssign`

Reply:
https://github.com/pyccel/pyccel/pull/1178/commits/aef0e90a9ae684bdb60f7a7e855bf8a7e9ee10de

Reply:
As C actually has augassign operators I think it would be good to keep these where possible. I would just use something similar to Fortran's [_print_AugAssign](https://github.com/pyccel/pyccel/blob/master/pyccel/codegen/printing/fcode.py#L2066-L2068) to handle the modulo case which can't be written as an augmented assign

Reply:
https://github.com/pyccel/pyccel/pull/1178/commits/815fed512c3e8f800f0a04aadcf334978beb913d